pawnShop/src/main/java/com/example/pawnShop/Service/ProductTypeServiceImp.java

```diff
@@ -0,0 +1,82 @@
+package com.example.pawnShop.Service;
+
+import com.example.pawnShop.Dto.Product.ProductTypeDto;
+import com.example.pawnShop.Factory.Contract.ProductFactory;
+import com.example.pawnShop.ManualMapper.Contract.ProductManualMapper;
+import com.example.pawnShop.Repository.ProductTypeRepository;
+import com.example.pawnShop.Service.Contract.ProductTypeService;
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@Transactional
+@Service
+@RequiredArgsConstructor
+public class ProductTypeServiceImp implements ProductTypeService {
+
+    private final ProductTypeRepository repository;
+    private final ProductManualMapper mapper;
+    private final ProductFactory factory;
+
+    @Override
+    public ProductTypeDto getProductTypeById(UUID id) {
+
+        //todo decide how to manage exceptions
+        var productType = repository.findById(id).orElseThrow();
```

Comment:
.orElseThrow(() -> new IllegalArgumentException
                        (String.format("Product has not been found in the database. ProductId: %s", ProductId)));